### PR TITLE
Properly escape custom props

### LINKED ISSUE #64: theme classnames (and css custom properties) are not properly escaped
### Describe the bug

I want to override the default tailwind `spacing` variables, that supports values like `1.5` and `0.5`, but tailwind-themer does not escape them, like tailwind vanilla does.

### Your minimal, reproducible example

https://codesandbox.io/p/github/cloud-walker/repro-tw-themer/main?workspace=%257B%2522activeFilepath%2522%253Anull%252C%2522openFiles%2522%253A%255B%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clgdw7o49000e3b6i6thrxuo6%2522%253A%257B%2522key%2522%253A%2522clgdw7o49000e3b6i6thrxuo6%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522key%2522%253A%2522clgdw7zco006m3b6ijqmx14qy%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522clgdw7xkn003f3b6i3twwmkgq%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clgdw7o49000e3b6i6thrxuo6%2522%252C%2522spacesOrder%2522%253A%255B%2522clgdw7o49000e3b6i6thrxuo6%2522%255D%252C%2522hideCodeEditor%2522%253Afalse%257D

### Steps to reproduce

Just inspect the output

### Expected behavior

the classnames and css custom properties should be escaped, like tailwind does

### How often does this bug happen?

Every time

### Screenshots or Videos

image

### Platform

- OS: MacOS
- Browser: Chrome latest

### tailwindcss-themer version

latest

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cloud-walker**
fixes #64 

I've tried taking advantage of the `api.e` function, but I've then discovered that it cannot be passed [here](https://github.com/RyanClementsHax/tailwindcss-themer/blob/b1a921575c55366f1d6de6bc1af36476fbbb03f3/src/index.ts#L61)

Probably tailwindcss package should expose the escape function directly or give the possibility to pass the api object to create the theme config 🤔

### CODE REVIEW

File: src/utils/themeUtils.spec.ts [hunk 1]
```diff
@@ -701,7 +702,7 @@ describe('themeUtils', () => {
           helpers
         )
       ).toEqual({
-        '--foo': 'thing'
+        [escape('--foo')]: 'thing'
```
Comment (RyanClementsHax) [inline src/utils/themeUtils.spec.ts:705]: The changes to add `escape` here and below aren't needed right? These varaiables should remain the same. Could we keep them the same to help assert we don't make regressions?

Also can we add a test explicitly expecting the escaping of the special characters? i.e. a test that verifies this fixes your issue?

**Review by RyanClementsHax — COMMENTED** [review]
Overall good work and very clean. Just left a couple of nit picks

File: src/utils/themeUtils.spec.ts (hunk 1 above)
Comment (cloud-walker) [reply src/utils/themeUtils.spec.ts]: You are right, my bad 🙂

File: src/index.spec.ts [hunk 2]
```diff
@@ -58,14 +58,14 @@ describe('multiThemePlugin', () => {
...
-          '--colors-primary': 'thing'
+          [escape('--colors-primary')]: 'thing'
         }
       })
       for (const theme of config.themes ?? []) {
         expect(api.addBase).toHaveBeenCalledWith({
           [`.escaped-${theme.name}`]: {
-            '--colors-primary': 'another',
-            '--colors-secondary': 'something'
+            [escape('--colors-primary')]: 'another',
```
[the remark this answers is not in the document]
Comment (cloud-walker) [reply src/index.spec.ts]: You are right, my bad 🙂

**Comment by cloud-walker** [comment]
Oh! I think I've putted the `escape` function because before I tried to use the escape function mocked (the one that adds `escaped-` prefix).

But then I changed direction 🤔

**Comment by cloud-walker** [comment]
Updated, let me know if I can do something else 🙂

File: src/utils/customPropUtils.ts [hunk 3]
```diff
@@ -1,5 +1,17 @@
 import { isColor, toRgb, withOpacity } from './colorUtils'
 
+/**
+ * Code copied from the tailwind official codebase
+ * @link https://github.com/tailwindlabs/tailwindcss/blob/fbbba6f67f73c3a4f9571649c3fc27006446d8f4/src/lib/regex.js#L70
```
Comment (RyanClementsHax) [inline src/utils/customPropUtils.ts:5]: Btw good job including a permalink here

**Review by RyanClementsHax — APPROVED** [review]

**Review by RyanClementsHax — CHANGES_REQUESTED** [review]
Oh sorry! One other thing I forgot (my fault). Could you add a commit using the commit message conventions listed here? https://github.com/RyanClementsHax/tailwindcss-themer/blob/main/CONTRIBUTING.md#commit-message-conventions

This allows the pipeline to autogenerate a release and automatically version bump the project. A new version won't be released after merging this pull request unless the pipeline sees a new commit matching this convention.

**Comment by cloud-walker** [comment]
Done, let me know if I did it right 🙂

**Comment by RyanClementsHax** [comment]
all good! thanks for your contribution!

**Pull request merged**